pkg/pprof/cell.go

```diff
@@ -0,0 +1,126 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package pprof
+
+import (
+	"errors"
+	"net"
+	"net/http"
+	_ "net/http/pprof"
+	"strconv"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/pflag"
+
+	"github.com/cilium/cilium/pkg/hive"
+	"github.com/cilium/cilium/pkg/hive/cell"
+)
+
+const (
+	// Pprof is the flag to enable the registration of pprof HTTP handlers
+	Pprof = "pprof"
+
+	// PprofAddress is the flag to set the address that pprof listens on
+	PprofAddress = "pprof-address"
+
+	// PprofPort is the flag to set the port that pprof listens on
+	PprofPort = "pprof-port"
+)
+
+type Server interface {
+	// Port returns the port at which the server is listening
+	Port() int
+}
+
+// Cell creates the cell for pprof, that registers its HTTP handlers to serve
+// profiling data in the format expected by the pprof visualization tool.
+var Cell = cell.Module(
+	"pprof",
+	"pprof HTTP handlers",
```

Comment:
I find talking about "HTTP handlers" here (and in cells.go) confusing as we're actually running an HTTP server and not providing handlers to some other HTTP server. I would instead title this as: "Exposes pprof runtime profiling over HTTP" or something like that.